localstack/services/sns/provider.py

```diff
@@ -1041,3 +1042,44 @@ def on_delete(self, request: Request) -> Response:
 
         store.sms_messages.clear()
         return Response("", status=204)
+
+
+class SNSServiceSubscriptionTokenApiResource:
+    """Provides a REST API for retrospective access to Subscription Confirmation Tokens to confirm subscriptions.
+    Those are not sent for email, and sometimes inaccessible when working with external HTTPS endpoint which won't be
+    able to reach your local host.
+
+    This is registered as a LocalStack internal HTTP resource.
+
+    This endpoint accepts:
+    - GET param `accountId`: selector for AWS account. If not specified, return fallback `000000000000` test ID
+    - GET param `region`: selector for AWS `region`. If not specified, return default "us-east-1"
+    - GET param `subscriptionArn`: filter for `subscriptionArn` resource in SNS
+    """
+
+    @route(sns_constants.SUBSCRIPTION_TOKENS_ENDPOINT, methods=["GET"])
+    def on_get(self, request: Request):
+        account_id = request.args.get("accountId", DEFAULT_AWS_ACCOUNT_ID)
+        region = request.args.get("region", AWS_REGION_US_EAST_1)
+        subscription_arn = request.args.get("subscriptionArn")
+        store: SnsStore = sns_stores[account_id][region]
```

Comment:
You're right, I followed the same logic as the previous one, because the `subscriptionArn` is optional, but I'm not so sure it should be. Maybe we could make it part of the route and not have those as parameters at all.
You just wouldn't fetch all the possible tokens. I think that'd be simpler, just a bit different from the previous ones but it makes sense. Will do that actually, if that's ok?